src/Riok.Mapperly/SyntaxProvider.Roslyn4.0.cs

```diff
@@ -12,6 +13,40 @@ internal static class SyntaxProvider
     private static readonly SymbolDisplayFormat _fullyQualifiedFormatWithoutGlobal =
         SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.OmittedAsContaining);
 
+    public static IncrementalValueProvider<ImmutableArray<Compilation>> GetNestedCompilations(
+        IncrementalGeneratorInitializationContext context
+    ) =>
+        context
+            .GetMetadataReferencesProvider()
+            .SelectMany((metadataReference, _) => new[] { (metadataReference as CompilationReference)?.Compilation }.ToImmutableArray())
+            .WhereNotNull()
+            .Collect();
```

Comment:
Couldn't this be simplified to
```csharp
context
            .GetMetadataReferencesProvider()
            .Select((metadataReference, _) => (metadataReference as CompilationReference)?.Compilation)
            .WhereNotNull()
            .Collect();
```

Or maybe even introduce a `OfType` extension method similar to the one of linq?
```csharp

context
            .GetMetadataReferencesProvider()
            .OfType<CompilationReference>()
            .Select((compilationReference, _) => compilationReference.Compilation)
            .Collect();
```

Reply:
Not as nice as I had imagined due to the incremental values provider not having covariance but IMO still easier to read than the previous approach, thanks for updating 😊